sonar-scanner-engine/src/main/java/org/sonar/scanner/scm/ScmPublisher.java

```diff
@@ -92,33 +91,36 @@ public void execute(SensorContext context) {
     }
   }
 
-  private List<InputFile> collectFilesToBlame() {
+  private List<InputFile> collectFilesToBlame(ScannerReportWriter writer) {
     if (configuration.forceReloadAll()) {
       LOG.warn("Forced reloading of SCM data for all files.");
     }
     List<InputFile> filesToBlame = new LinkedList<>();
-    for (InputFile f : fs.inputFiles(fs.predicates().all())) {
+    for (InputFile f : componentStore.inputFiles()) {
+      DefaultInputFile inputFile = (DefaultInputFile) f;
+      if (!inputFile.publish()) {
+        continue;
+      }
       if (configuration.forceReloadAll() || f.status() != Status.SAME) {
         addIfNotEmpty(filesToBlame, f);
       } else {
         // File status is SAME so that mean fileData exists
-        FileData fileData = projectRepositories.fileData(projectDefinition.getKeyWithBranch(), f.relativePath());
+        FileData fileData = projectRepositories.fileData(inputModule.definition().getKeyWithBranch(), f.relativePath());
         if (StringUtils.isEmpty(fileData.revision())) {
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png 'Severity: MAJOR') NullPointerException might be thrown as 'fileData' is nullable here [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3AS2259)